postgresql-ballerina/client.bal

```diff
@@ -0,0 +1,264 @@
+
+type ClientConfiguration record {|
+    string host;
+    int port;
+    string? user;
+    string? password;
+    string? database;
+    Options? options;
+    sql:ConnectionPool? connectionPool;
+|};
+
+# PostgreSQL database options.
+#
+# + ssl - SSL Configuration to be used
+# + connectTimeout - Timeout to be used when connecting to the postgresql server.
+# + socketTimeout - Socket timeout during the read/write operations with postgresql server,
+#                            0 means no socket timeout.
+# + loginTimeout - Specify how long to wait for establishment of a database connection.The timeout 
+#                           is specified in seconds.
+# + rowFetchSize - Determine the number of rows fetched in ResultSet by one fetch with trip to the database.
+# + dbMetadataCacheFields - Specifies the maximum number of fields to be cached per connection.
+#                           A value of 0 disables the cache.
+# + dbMetadataCacheFieldsMiB - Specifies the maximum size (in megabytes) of fields to be cached per connection. 
+#                            A value of 0 disables the cache.
+# + prepareThreshold - Determine the number of PreparedStatement executions required before switching over to use 
+#                            server side prepared statements.
+# + preparedStatementCacheQueries - Determine the number of queries that are cached in each connection.
+# + preparedStatementCacheSize - Determine the maximum size (in mebibytes) of the prepared queries.
+# + cancelSignalTimeout - Cancel command is sent out of band over its own connection, so cancel 
+#                                 message can itself get stuck. So the timeout seconds for that.
+# + tcpKeepAlive - Enable or disable TCP keep-alive probe.
+# + loggerLevel - Logger level of the driver. Allowed values: OFF, DEBUG or TRACE.
+# + loggerFile - File name output of the Logger. This parameter should be use together with loggerLevel.
+# + logUnclosedConnections - Enable or disable Log details about unclosed connections.
+# + binaryTransfer - Use binary format for sending and receiving data if possible
+
+public type Options record {|
+  SSLConfig ssl = {};
+  decimal connectTimeout?;
+  decimal socketTimeout?;
+  decimal loginTimeout?;
+  int rowFetchSize?;
+  int dbMetadataCacheFields?;
+  int dbMetadataCacheFieldsMiB?;
+  int prepareThreshold?;
+  int preparedStatementCacheQueries?;
+  int preparedStatementCacheSize?;
+  decimal cancelSignalTimeout?;
+  boolean tcpKeepAlive?;
+  LoggerLevel loggerLevel?;
+  string loggerFile?;
+  boolean logUnclosedConnections?;
+  boolean binaryTransfer?;
+|};
+
+# Possible values for SSL mode.
+# 
+public enum SSLMode {
+   PREFER,
+   REQUIRE,
+   DISABLE,
+   ALLOW,
+   VERIFY_CA = "VERIFY-CA",
+   VERIFY_FULL = "VERIFY-FULL"
+}
+
+# Possible values for Logger Level Connection Parameter.
+# 
+public enum LoggerLevel {
```

Comment:
Let's not support Logger

Reply:
Removed. Commit id is 79ccf2074765e4298ea0788e22a3bc3e33604b15